crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
Before this change, when an SSL object is dup'd, new ex_data is created (i.e. the new callback is invoked), and then the data is duplicated (i.e. the dup callback is invoked). With this set of lines, the data added by the new callback is leaked/lost, and the dup callback may consider it an error if there's no where to put the data. This occurs only on the last set of callbacks. So, if you have two sets of ex_data callbacks added to an SSL, then the last one that was added will have different behavior than ones before it.